Document SystemRegistrationContext; enforce inheritance; clean includes (#146)
RobotecAI/o3de-genai-gems: README.md

- `GenAIVendorBundle`: a vendor Gem that implements communication with [Anthropic](https://docs.anthropic.com/en/docs/welcome) service for running `Claude` model and [Ollama toolset](https://github.com/ollama/ollama) that allows getting up and running various language models locally.
- `GenAIAmazonBedrockVendorBundle Gem`: a vendor Gem that implements communication with [Amazon Bedrock service](https://aws.amazon.com/bedrock/) and that offers a choice of high-performing models from leading AI companies. This gem is dependent on the `GenAIVendorBundle` Gem and allows for communication with the `Claude` model using the `Amazon Bedrock` service.

Additionally, the `GenAIMock` vendor Gem is provided to respond in user-configurable responses. This Gem is meant for cost-effective and fast tests as well as a sample implementation of a vendor Gem.

To learn more about the available vendors and find a guide for the configurations, see the [vendors'](./docs/vendors.md) documentation. There is also a page describing how to add a [new vendor Gem](./docs/newVendor.md). 

### Feature Gems

> Feature Gems use Generative AI to deal with a certain problem in your workflow. For example, a _scene generation feature Gem_ might be responsible for creating a scene based on the user's prompt. Internally, it could connect to the asset processor to discover available assets and to the O3DE Editor to learn about the existing scene layout. Next, it could combine all necessary information and send a query to the AI. Based on the response, it could call some existing O3DE buses to add/move/remove the elements of the scene. The communication between the Generative AI and the feature is independent from the O3DE, hence the feature could iterate multiple times with the AI to find the best solution for the user's prompt. Finally, the feature should inform the user about the completion and send an answer to the prompt.

Reviewer: > The communication between the Generative AI and the feature is independent from the O3DE.  

I find it hard to understand this sentence. Maybe rephrasing it would be better.
```suggestion
Feature Gems use Generative AI to deal with a certain problem in your workflow. For example, a _scene generation feature Gem_ might be responsible for creating a scene based on the user's prompt. Internally, it could connect to the asset processor to discover available assets and to the O3DE Editor to learn about the existing scene layout. Next, it could combine all necessary information and send a query to the AI. Based on the response, it could call some existing O3DE buses to add/move/remove the elements of the scene. Communication with O3DE is not request-response based, hence the feature could iterate multiple times with the AI to find the best solution for the user's prompt. Finally, the feature should inform the user about the completion and send an answer to the prompt.
```

Author: This change is not related to this PR, but I pushed it.
